opensrp-giz-malawi/src/main/java/org/smartregister/giz/view/NavigationMenu.java

```diff
@@ -0,0 +1,333 @@
+        // update all actions
+        mPresenter.refreshLastSync();
+    }
+
+    private void registerNavigation(Activity parentActivity) {
+        if (recyclerView != null) {
+            navigationOptions = mPresenter.getOptions();
+            if (navigationAdapter == null) {
+                navigationAdapter = new NavigationAdapter(navigationOptions, parentActivity);
+            }
+
+            RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(parentActivity);
+            recyclerView.setLayoutManager(mLayoutManager);
+            recyclerView.setItemAnimator(new DefaultItemAnimator());
+            recyclerView.setAdapter(navigationAdapter);
+        }
+    }
+
+    private void registerLogout(final Activity parentActivity) {
+        mPresenter.displayCurrentUser();
+        tvLogout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                logout(parentActivity);
+            }
+        });
+    }
+
+    private void registerSync(final Activity parentActivity) {
+
+        TextView tvSync = rootView.findViewById(R.id.tvSync);
+        ivSync = rootView.findViewById(R.id.ivSyncIcon);
+        syncProgressBar = rootView.findViewById(R.id.pbSync);
+
+        View.OnClickListener syncClicker = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(parentActivity, parentActivity.getResources().getText(R.string.action_start_sync),
+                        Toast.LENGTH_SHORT).show();
+                mPresenter.sync(parentActivity);
+            }
+        };
+
+
+        tvSync.setOnClickListener(syncClicker);
+        ivSync.setOnClickListener(syncClicker);
+
+        refreshSyncProgressSpinner();
+    }
+
+    private void registerLanguageSwitcher(final Activity context) {
+        final TextView tvLang = rootView.findViewById(R.id.tvLang);
+        Locale current = context.getResources().getConfiguration().locale;
+        tvLang.setText(StringUtils.capitalize(current.getDisplayLanguage()));
+    }
+
+    private void registerDeviceToDeviceSync(@NonNull final Activity activity) {
+        rootView.findViewById(R.id.rlIconDevice)
+                .setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        startP2PActivity(activity);
+                    }
+                });
+    }
+
+    protected void refreshSyncProgressSpinner() {
+        if (SyncStatusBroadcastReceiver.getInstance().isSyncing()) {
+            syncProgressBar.setVisibility(View.VISIBLE);
+            ivSync.setVisibility(View.INVISIBLE);
+        } else {
+            syncProgressBar.setVisibility(View.INVISIBLE);
+            ivSync.setVisibility(View.VISIBLE);
+        }
+    }
+
+    public void startP2PActivity(@NonNull Activity activity) {
+        if (PermissionUtils.isPermissionGranted(activity
```

Comment:
You do not need to check for the permissions currently

Reply:
Does the lib check for the permissions? Why don't I need to?

Reply:
Yes, it checks for the permissions